libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/ConfigurationChannelProvider.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Linq;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Bot.Builder.BotFramework
+{
+    /// <summary>
+    /// Channel provider which uses <see cref="Microsoft.Extensions.Configuration.IConfiguration"/> to lookup the channel service property.
+    /// </summary>
+    /// <remarks>
+    /// This will populate the <see cref="SimpleChannelProvider.ChannelService"/> from a configuration entry with the key of <see cref="ChannelServiceKey"/>.
+    ///
+    /// NOTE: if the keys are not present, a <c>null</c> value will be used.
+    /// </remarks>
+    public sealed class ConfigurationChannelProvider : SimpleChannelProvider
+    {
+        /// <summary>
+        /// The key for ChannelService.
+        /// </summary>
+        public const string ChannelServiceKey = "ChannelService";
+        
+        public ConfigurationChannelProvider(IConfiguration configuration)
+        {
+            this.ChannelService = configuration.GetSection(ChannelServiceKey)?.Value;
+        }
```

Comment:
Should this throw if the value is not present?

Reply:
Well, blank or null means we use "regular" channel service values. There isn't an exception when there isn't an AppId or Password, right? I could throw and say "only use this if you are connecting to gov or a private DirectLine and here is the constant to use"...what do you think?

Reply:
No, the value can be null/empty string and that is 'Public Azure' as this is what we have now. The reason I prefer it this way is that I want people to use the same bot code for a bot deployment in Gov and non-Gov where the only change is to the app.settings. This means we have to be ok with putting nothing in there but still trying to read the property.

Reply:
Can you put this into a code comment here? That way people can see it. 

---
In reply to: [214128651](https://github.com/Microsoft/botbuilder-dotnet/pull/890#discussion_r214128651) [](ancestors = 214128651)